src/backend/gl/Cargo.toml

```diff
@@ -17,8 +17,9 @@ build = "build.rs"
 name = "gfx_backend_gl"
 
 [features]
-default = []
+default = ["wgl", "surfman", "surfman-x11"]
```

Comment:
It won't fail. CI passes on all platforms with all of those features enabled. That was part of point of the latest refactoring. The setup is that you could enable any features you want and it won't error on any platform because the features are properly gated. So even if you enable `wgl` on Linux or `surfman-x11` on MacOS, it will just ignore those because they aren't supported on that platform. That way we don't have to dynamically enable or disable features based on the target platform. We can just put whatever features we want to be enabled across all platforms in the default and users can happily build anywhere without specifying anything.

Because they are features though, if you want to build on Linux without X11 you can still do `cargo build --no-default-features --features surfman,wgl` and it will build it with wgl on windows, but when building on Linux it will use surfman without X11.

Reply:
I wonder how big these dependencies are. In Gecko, we will not need to `wgl` for sure, for example, we wouldn't want to keep that in tree of dependencies.
In general, gfx-backend-gl users will unlikely be very interested in `wgl`, given the availability of the other backends on Windows. Maybe we could at least not make it a default?

Reply:
If you are using `gfx-backend-gl` on Windows and you don't provide the `wgl` feature or the `glutin` feature, it will end up compiling in Dummy mode and not doing anything. If that were the case for your application I would think you would just not include `gfx-backend-gl` at all on  that platform by using target specific dependencies.

In other words if we don't default either `glutin` or `wgl`, the default operation of `gfx-backend-gl` on windows would be to do nothing which doesn't seem right to me.

Reply:
Right, I see. I think it's fine. The downside for having `wgl` is that you pay for this dependency to be included in the tree even though you aren't going to use it ever.

let's say, if there is a user who needs GL on windows and Linux, they will just enable `wgl` feature unconditionally. Others will not need to know about it.

Reply:
I'm still a little confused. When would you include `gfx-backend-gl` on Windows if you weren't going to use it on Windows? The `wgl` dependencies aren't included when you are building on Linux.

Would the problem situation be when a user includes `gfx-backend-gl` for a project, only intending to use GL on Linux, and then accidentally including the `wgl` dependencies when building for Windows, even though there were actually using something like the `dx11` backend for Windows?

Reply:
We wouldn't include gfx-backend-gl on Windows for `wgpu`, but the way your PR is set up right now, we'd still technically depend on `wgl`. Which means `wgl` and its dependencies will be erroneously vendored in Gecko third party sources.

Reply:
In that case I think we would just do this in `wgpu`:

```toml
[target.'cfg(unix, not(target_os = "ios")'.dependencies]
gfx-backend-gl = "0.5"
```
Because you aren't including `gfx-backend-gl` on Windows, the `wgl` dependencies ( which is actually just the `winapi` crate ) will never be included into the tree because the `winapi` crate is only brought in by `gfx-backend-gl` on windows:

```toml
[target.'cfg(windows)'.dependencies]
winapi = { version = "0.3", features = ["libloaderapi", "windef", "wingdi", "winuser"] }
```

So if you don't include `gfx-backend-gl` on windows, you won't ever include `winapi`.

Reply:
Actually, if it's just winapi, then we are totally fine - it's already vendored on Windows.
:)